streamlit_app.py

```diff
@@ -5,12 +5,54 @@
 import requests
 import en_core_web_sm
 import geocoder
+import base64
+import json
 
 # Load environment variables at the top to avoid errors
 load_dotenv()
 
 # I was not able to get the key to work loading from the env why?
 API_WEATHER_KEY = os.getenv("API_WEATHER_KEY")
+#Spotify API keys
+client_id = os.getenv("CLIENT_ID")
+client_secret = os.getenv("CLIENT_SECRET")
+refresh_token = os.getenv("REFRESH_TOKEN")
+
+#Spotify API access to tokens
```

Comment:
Try to be more detailed on the explanation of the function. What is it doing exactly? 

Docstrings help the reading of code. Checkout this [python function docstring example](https://www.programiz.com/python-programming/docstrings#:~:text=2.%20Docstrings%20for%20Python%20Functions) format.

Try to follow this format in the rest of the functions